### DISCUSSION TITLE
Geo based feature flags

### DISCUSSION THREAD

**Discussion by farzinnasiri** [Unleash/unleash#7615] [Q&A] [answered]
Hi, 
we recently discovered unleashed and deployed an instance on our servers to explore the product. 
one important item for our company is the ability to activate features based on the user's location, for instance, different countries.
I see that Unleash does not have this option out of the box, but is there any way to implement it?

Thank you

**Comment by malte-baumann** [comment]
I suggest you have a look at custom activation strategies if you have access to the location data.
https://docs.getunleash.io/reference/custom-activation-strategies

It then varies what SDK you use.

**Reply by malte-baumann** [reply, marked as answer]
Update: Have also a look at strategy constraints. https://docs.getunleash.io/reference/strategy-constraints 
They do have an example with regions in there as well.

**Reply by farzinnasiri** [reply]
Aren't custom activation strategies deprecated?

**Comment by farzinnasiri** [comment]
<img width="783" alt="Screenshot 2024-07-23 at 6 49 49 PM" src="https://github.com/user-attachments/assets/c0b3b874-9eb9-4580-9f74-3f919148736c">
<img width="668" alt="Screenshot 2024-07-23 at 6 50 09 PM" src="https://github.com/user-attachments/assets/1ff609ab-5429-453c-87f3-1c37a3cb197c">

furthermore I don't see the option to filter users based on the region in my deployment, but the option is said to be available in the docs:
[here](https://docs.getunleash.io/reference/strategy-constraints)

**Reply by malte-baumann** [reply]
You can create your own context field. See https://docs.getunleash.io/reference/unleash-context#custom-context-fields
